mongomock/collection.py

```diff
@@ -1417,19 +1418,44 @@ def aggregate(self, pipeline, **kwargs):
                                             new_doc = False
                                             break
 
-                                    current_val = doc_dict.get(field, 0)
                                     from_field = key.replace('$', '')
                                     if func == "$sum":
+                                        current_val = doc_dict.get(field, 0)
                                         for doc in group_list:
```

Comment:
`doc_dict[field] = doc_dict.get(field,0) + sum([doc.get(from_field,0) for doc in group_list])`